### PR TITLE
class library: make control set atomic

### PR DESCRIPTION
**Pull request by LFSaw**
this fix by @telephon fixes an annoying behaviour in NodeProxies:

```
Ndef(\sine).clear


(
Ndef(\sine, {SinOsc.ar(\freq.kr(100)!2).tanh});
Ndef(\sine).set(\freq, 200);
)

-> Ndef('sine')
FAILURE IN SERVER /n_set Node 1010 not found
```

Previous behaviour:
```
1. set source: create a bundle that:
1a. sends synth def
1b. waits for completion, then sends synth with args
2. set controls: create a bundle that:
2a. sets group

temporal order will be: 1a. 2a. 1b.
```

now, it does it different... (maybe ask @telephon for details..)
I use this branch extensively for a week now and think that it works perfectly.

### CODE REVIEW

**Comment by telephon** [comment]
The branch breaks:


````
Ndef(\x, { |freq| SinOsc.ar(freq) }); 
Ndef(\x).send;
````

crashes the interpreter

**Comment by telephon** [comment]
OK, fixed. So please test it extensively!

**Comment by telephon** [comment]
@crucialfelix – because e wrote the ```OSCBundle``` stuff together back in the old days: can you have a brief look at the change?

File: SCClassLibrary/JITLib/ProxySpace/NodeProxy.sc [hunk 1]
```diff
@@ -746,9 +745,9 @@ NodeProxy : BusPlug {
 
 	// bundle: apply the node map settings to the entire group
 	sendAllToBundle { | bundle, extraArgs |
-		extraArgs = nodeMap.asOSCArgArray ++ extraArgs.value.asOSCArgArray;
+		var args = Thunk { nodeMap.asOSCArgArray ++ extraArgs.value.asOSCArgArray };
```
Comment (crucialfelix) [inline SCClassLibrary/JITLib/ProxySpace/NodeProxy.sc]: better style IMO would be:

```supercollider
Thunk({ ... })
```
which makes it clear that you are creating a Thunk object.

**Comment by crucialfelix** [comment]
seems okay. its always better not to complicate the core classes, but there is already a bunch of junk in OSCBundle

**Pull request merged**